superset-frontend/src/dashboard/actions/dashboardState.js

```diff
@@ -194,52 +196,176 @@ export function saveDashboardRequest(data, id, saveType) {
     const serializedFilters = serializeActiveFilterValues(getActiveFilters());
     // serialize filter scope for each filter field, grouped by filter id
     const serializedFilterScopes = serializeFilterScopes(dashboardFilters);
+    const {
+      certified_by,
+      certification_details,
+      css,
+      dashboard_title,
+      owners,
+      roles,
+      slug,
+    } = data;
+
+    // making sure the data is what the backend expects
+    const cleanedData = {
+      ...data,
+      certified_by: certified_by || '',
+      certification_details:
+        certified_by && certification_details ? certification_details : '',
+      css: css || '',
+      dashboard_title: dashboard_title || t('[ untitled dashboard ]'),
+      owners: owners && owners.length ? owners.map(o => (o.id ? o.id : o)) : [],
```

Comment:
`owners && owners.length` -> `ensureIsArray(owners)`
`owners.map(o => (o.id ? o.id : o))` - I assume that means that owner can either be a string or an object with id? Maybe we could use a type guard to make it a bit clearer?